pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -0,0 +1,174 @@
+"""Plugin for constructor definitions."""
+from __future__ import annotations
+
+from ast import arguments
+from dataclasses import dataclass
+import re
+
+import astroid
+from pylint.checkers import BaseChecker
+from pylint.interfaces import IAstroidChecker
+from pylint.lint import PyLinter
+
+
+@dataclass
+class TypeHintMatch:
+    """Class for pattern matching."""
+
+    module_filter: re.Pattern
+    function_name: str
+    # TODO: add the correct message
+    arg_types: dict[int, str]
+    return_type: str | None
+    disabled: bool = False
+
+
+_METHOD_MATCH: list[TypeHintMatch] = [
+    TypeHintMatch(
+        module_filter=re.compile(r"^homeassistant.components.[a-zA-Z_]+$"),
```

Comment:
Maybe make this a constant ?

Reply:
I've created a few constants now. This should be better.